pkg/server/metrics.go

```diff
@@ -0,0 +1,162 @@
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"sigs.k8s.io/release-utils/version"
+)
+
+type metrics struct {
+	reg           *prometheus.Registry
+	serverMetrics *grpc_prometheus.ServerMetrics
+	// metrics
+	newHashedRekordEntries prometheus.Counter
+	newDsseEntries         prometheus.Counter
+	httpLatency            *prometheus.HistogramVec
+	httpRequestsCount      *prometheus.CounterVec
+}
+
+// Metrics provides the singleton metrics instance
+func getMetrics() *metrics {
+	return _initMetricsFunc()
+}
+
+var _initMetricsFunc = sync.OnceValue[*metrics](func() *metrics {
+	m := metrics{
+		reg:           prometheus.NewRegistry(),
+		serverMetrics: grpc_prometheus.NewServerMetrics(),
+	}
+	m.reg.MustRegister(m.serverMetrics)
+
+	f := promauto.With(m.reg)
+
+	m.newHashedRekordEntries = f.NewCounter(prometheus.CounterOpts{
+		Name: "rekor_new_hashedrekord_entries",
+		Help: "The total number of new dsse log entries",
+	})
+
+	m.newDsseEntries = f.NewCounter(prometheus.CounterOpts{
+		Name: "rekor_new_dsse_entries",
+		Help: "The total number of new dsse log entries",
+	})
+
+	m.httpLatency = f.NewHistogramVec(prometheus.HistogramOpts{
+		Name: "rekor_http_api_latency",
+		Help: "API Latency on HTTP calls",
+	}, []string{"code", "method"})
+
+	m.httpRequestsCount = f.NewCounterVec(prometheus.CounterOpts{
+		Name: "rekor_http_requests_total",
+		Help: "Count all HTTP requests",
+	}, []string{"code", "method"})
+
+	// TODO: add metrics from rekor v1 (anything but Counter appears to need to be a pointer)
+
+	_ = f.NewGaugeFunc(
+		prometheus.GaugeOpts{
+			Namespace: "rekor_v2",
+			Name:      "build_info",
+			Help:      "A metric with a constant '1' value labeled by version, revision, branch, and goversion from which rekor was built.",
+			ConstLabels: prometheus.Labels{
+				"version":    version.GetVersionInfo().GitVersion,
+				"revision":   version.GetVersionInfo().GitCommit,
+				"build_date": version.GetVersionInfo().BuildDate,
+				"goversion":  version.GetVersionInfo().GoVersion,
+			},
+		},
+		func() float64 { return 1 },
+	)
+	return &m
+})
+
+type httpMetrics struct {
+	*http.Server
+	serverEndpoint string
+}
+
+func newHTTPMetrics(_ context.Context, config *HTTPConfig) *httpMetrics {
+	mux := http.NewServeMux()
+	mux.Handle("/", promhttp.HandlerFor(getMetrics().reg, promhttp.HandlerOpts{}))
+
+	// TODO: configure https connection preferences (time-out, max size, etc)
+
+	endpoint := config.HTTPMetricsTarget()
+	return &httpMetrics{
+		Server: &http.Server{
+			Addr:    endpoint,
+			Handler: mux,
+
+			ReadTimeout:       60 * time.Second,
+			ReadHeaderTimeout: 60 * time.Second,
+			WriteTimeout:      60 * time.Second,
+			IdleTimeout:       config.idleTimeout,
+		},
+		serverEndpoint: endpoint,
+	}
+}
+
+func (hp *httpMetrics) start(wg *sync.WaitGroup) {
```

Comment:
This is very much like all the other servers, although I was looking at `github.com/oklog/run` to simplify our signal handling and server shutdowns (I dunno).